gh-101100: Fix some Sphinx reference warnings in whatsnew/2.6.rst

rowanbudge opened the pull request:
For gh-101100 - Fix all Sphinx reference warnings in the documentation #101100

Fix some of the issues in whatsnew/2.6.rst 

Part of sprints at PyConUK 25

<!-- readthedocs-preview cpython-previews start -->
----
📚 Documentation preview 📚: https://cpython-previews--139236.org.readthedocs.build/

<!-- readthedocs-preview cpython-previews end -->

rowanvil committed:
Ignore future_builtins

rowanvil committed:
Ignore sys.py3kwarning

rowanvil committed:
Make example code reference literals for DatabaseConnection and Drawable

Also add a working link for abc.abstractmethod while there

rowanvil committed:
fix contextlib references

rowanvil committed:
fix multiprocessing references

rowanvil committed:
Fix New I/O Library refernces

rowanvil committed:
Fix Abstract Base Classes references

rowanvil committed:
Ignore APPDATA

StanFromIreland commented on Doc/whatsnew/2.6.rst:
```diff
@@ -1172,17 +1172,17 @@ Let's say you have a particular class and wish to know whether it supports
-and :meth:`update`?  Iterating over the object with :func:`iter`?
+Or that the object will have :meth:`!keys`, :meth:`!values`, and :meth:`!items`
+methods?  What about the iterative variants  such as :meth:`!iterkeys`?  :meth:`!copy`
+and :meth:`!update`?  Iterating over the object with :func:`!iter`?
```
```:func:`iter`​``` is perfectly fine?

AA-Turner replied:
Discussed at sprints. I advised Rowan to take this approach.

StanFromIreland commented on Doc/whatsnew/2.6.rst:
```diff
@@ -1038,47 +1038,47 @@ operations.
-  and :meth:`close`.
+* :class:`~io.RawIOBase` defines raw I/O operations: :meth:`~io.RawIOBase.read`,
+  :meth:`~io.RawIOBase.readinto`,
+  :meth:`~io.RawIOBase.write`, :meth:`~io.IOBase.seek`, :meth:`~io.IOBase.tell`, :meth:`~io.IOBase.truncate`,
```
Please wrap lines to 79 characters.

rowanvil replied:
I've wrapped at 79 lines in all the places already edited by this PR that I could spot

StanFromIreland replied:
Thanks, however, it introduced a lot of trailing white space causing the lint CI to fail.

rowanvil replied:
Yep aware and working on it

rowanvil replied:
Checks now pass

hugovk replied:
On the other hand, rewrapping _all_ edits makes it harder to review the actual changes in the diff, so it's a balance. (There's some discussion in https://github.com/python/devguide/issues/1018.)

No need to unwrap though.

Also please avoid force-pushing -- everything gets squashed into a single commit at the end, and separate commits are easier to review. Thanks!

StanFromIreland commented on Doc/whatsnew/2.6.rst:
```diff
@@ -307,9 +307,9 @@ The :mod:`threading` module's locks and condition variables  also support the
-to save and restore the current decimal context, which encapsulates the desired
-precision and rounding characteristics for computations::
+The :func:`decimal.localcontext` function in the :mod:`decimal` module makes 
+it easy to save and restore the current decimal context, which encapsulates 
```
```suggestion
The :func:`decimal.localcontext` function in the :mod:`decimal` module makes
it easy to save and restore the current decimal context, which encapsulates
```
E.g. above; Running `pre-commit` will fix them all at once.

rowanvil replied:
I've got that running now, thanks for the advice :)

rowanvil committed:
Review Changes - Wrap lines v2

now with pre-commit check to remove whitespace

rowanvil committed:
Fix Unexpected indentation Warning

Hopefully also resolve Block quote ends without a blank line; unexpected unindent.

hugovk reviewed (approved):
Thank you!

AA-Turner reviewed (commented):
Is this PR intended to resolve all warnings in the file? It's OK if not, just trying to get a sense of intended scope for the review.

A

AA-Turner committed:
Tweaks
